libgraph/include/katana/PropertyGraph.h

```diff
@@ -279,12 +279,22 @@ class KATANA_EXPORT PropertyGraph {
       const katana::RDGManifest& rdg_manifest,
       const katana::RDGLoadOptions& opts, katana::TxnContext* txn_ctx);
 
-  /// Make a projected graph from a property graph. Shares state with
-  /// the original graph.
-  static std::unique_ptr<PropertyGraph> MakeProjectedGraph(
+  [[deprecated]] static std::unique_ptr<PropertyGraph> MakeProjectedGraph(
       const PropertyGraph& pg, const std::vector<std::string>& node_types,
       const std::vector<std::string>& edge_types);
 
+  /// Make a projected graph from a property graph. Shares state with
+  /// the original graph.
+  static Result<std::unique_ptr<PropertyGraph>> MakeProjectedGraph(
+      const PropertyGraph& pg, const std::vector<std::string>* node_types,
+      const std::vector<std::string>* edge_types);
```

Comment:
Could we pass `std::optional<std::vector>` instead, since we're improving the API?